pkg/v2/tkr/webhook/infra-machine/fieldsetter/fieldsetter.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package fieldsetter defines methods to set the fields of <Infra>Machine resource  .
+package fieldsetter
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	// corev1 "k8s.io/api/core/v1"
+	"github.com/go-logr/logr"
+	"gopkg.in/yaml.v2"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const (
+	osImageRefAnnotationKey = "run.tanzu.vmware.com/os-image-ref"
+)
+
+// FieldSetter mutates <infra>Machine
+type FieldSetter struct {
+	decoder      *admission.Decoder
+	Log          logr.Logger
+	FieldPathMap map[string]string
+}
+
+// Handle will handle the infra admission request
+func (fs *FieldSetter) Handle(ctx context.Context, req admission.Request) admission.Response { // nolint:gocritic
```

Comment:
We should _always_ explain why we're using `// nolint` comments.

But actually, we should always try to fix linter errors first.

Reply:
Sure. This linter error is suggesting to use pointer parameter for `req` which is defined by standard signature for webhook registered callback handler. I will update the PR with this comment, but open to any suggestions to fix it.

Reply:
Thanks!